pkg/cloud/services/ec2/launchtemplate.go

```diff
@@ -34,16 +34,12 @@ import (
 
 // GetLaunchTemplate returns the existing LaunchTemplate or nothing if it doesn't exist.
 // For now by name until we need the input to be something different
-func (s *Service) GetLaunchTemplate(id string) (*expinfrav1.AWSLaunchTemplate, error) {
-	if id == "" {
```

Comment:
Wouldn't it be useful to keep that guardrail, just in case the AWSMachinePool name is empty (i.e. AMP doesn't exist for whatever reason)?

Reply:
On a second thought, nah, disregard.